_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp

```diff
@@ -5104,6 +5104,49 @@ mfxU64 make_def_back_color_yuv(mfxU16 bit_depth)
         ;
 };
 
+static
+mfxU64 get_background_color(const mfxVideoParam & videoParam)
+{
+    mfxU64 argb_back_color = 0xffff000000000000;
+    mfxU32 targetFourCC = videoParam.vpp.Out.FourCC;
+
+    switch (targetFourCC)
+    {
+        case MFX_FOURCC_NV12:
+        case MFX_FOURCC_YV12:
+        case MFX_FOURCC_NV16:
+        case MFX_FOURCC_YUY2:
+        case MFX_FOURCC_AYUV:
+            argb_back_color = make_def_back_color_yuv(8);
+            break;
+        case MFX_FOURCC_P010:
+#if (MFX_VERSION >= 1027)
+        case MFX_FOURCC_Y210:
+        case MFX_FOURCC_Y410:
+#endif
+        case MFX_FOURCC_P210:
+            argb_back_color = make_def_back_color_yuv(10);
+            break;
+        case MFX_FOURCC_RGB4:
+        case MFX_FOURCC_BGR4:
+            for (mfxU32 i = 0; i < videoParam.NumExtParam; i++)
+            {
+                if (videoParam.ExtParam[i]->BufferId == MFX_EXTBUFF_VPP_COMPOSITE)
+                {
+                    mfxExtVPPComposite *extComp = (mfxExtVPPComposite *) videoParam.ExtParam[i];
+                    argb_back_color = ((mfxU64) 0xff << 48) |
+                                      ((mfxU64) VPP_RANGE_CLIP(extComp->R, 0, 255) << 32) |
+                                      ((mfxU64) VPP_RANGE_CLIP(extComp->G, 0, 255) << 16) |
+                                      ((mfxU64) VPP_RANGE_CLIP(extComp->B, 0, 255) << 0);
```

Comment:
add ```break;``` to this loops body, since I believe there could be only one such buffer

Reply:
Added. There will be other buffer type with background color, but in next commits.